features/ei-analytics-feature/org.wso2.analytics.solutions.ei.analytics.feature/src/main/resources/siddhi-files/EI_Analytics_StatApp.siddhi

```diff
@@ -90,6 +95,11 @@ from ConfigEntryEventStream
 select meta_tenantId as metaTenantId, hashcode, entryName, configData, eventTimestamp() as eventTimestamp
 update or insert into ConfigEntryTable on ConfigEntryTable.hashcode == hashcode;
 
+-- add all the componentNames and componentTypes to ComponentNameTable, if the combination is unique
+from DecompressedEventStream[NOT ((ComponentNameTable.componentName == componentName AND ComponentNameTable.componentType == componentType) IN ComponentNameTable)]
+select componentName, componentType
+insert into ComponentNameTable;
```

Comment:
do an update/insert.. that way we dont have to check if it is already  there

Reply:
@isharailanga, what is the use case here, do you need keep the unique values in the table?
If that is the case then we dont need to update the table and we can just check the contains and insert(). But if you need to update the records, then the update or insert will be fast then contain check.

Reply:
@isharailanga Seems I was wrong and the way you sent the PR will be faster than the update or insert. Lets revert the last commit

Reply:
Fixed in 92ed523702f434f00b5e7a13623cc470edff0b94